src/asset/mod.rs

```diff
@@ -93,10 +160,20 @@ impl AnimationClip2D {
             ));
         }
 
+        let events = events.unwrap_or_default();
+        let max_event_frame = events.keys().max().cloned().unwrap_or(0);
+        if max_event_frame > keyframes_len {
```

Comment:
This should compare >=, the last frame will not trigger any events. If you don't want to do that, it should at the very least give a `warn!` that the event won't trigger.

Reply:
Good catch!

Reply:
Btw, with this change your trickfilm file does not work anymore. 
For example you have an animation with Range(0,7) and you define an event for frame 7, but the range is [0,7) (aka [0,6]) and the way event frames are specified is starting at 0.
So either your animation itself is too short or your event should trigger on frame 6 instead of 7.

Reply:
Yeah I only had it on the last frame to test whether it will fire or not, the whole reason I added the warn! was to confirm that. I already changed the events accordingly :+1: